src/Attribute.cpp

```diff
@@ -0,0 +1,70 @@
+#if HAVE_CONFIG_H
+#  include <config.h>
+#endif
+
+#ifdef WIN32
+#prattr_genma warning(disable : 4786)   /* Disable annoying warning messages */
+#endif
+
+#include "random.h"
+#include "OutputMgr.h"
+#include "Attribute.h"
+
+void
+AttributeGenerator::GenerateAllAttributes(std::ostream &out)
+{
+	bool attr_emitted = false;
+	string attr_generated = "";
+	vector<Attribute*>::iterator itr;
+	for(itr = attributes.begin(); itr < attributes.end(); itr++){
+		attr_generated = (*itr)->OutputAttributes();
+		if(!attr_emitted && attr_generated != ""){
+			out << " __attribute__((" << attr_generated;
+			attr_emitted = true;
+		}
+		else if(attr_generated != "")
+			out << ", " << attr_generated;
+	}
+	if(attr_emitted){
+		out << "))";
+		attr_emitted = false;
+	}
+}
+
+Attribute::Attribute(string name, int prob)
+	:  attribute(name), attribute_probability(prob), attr_generate("")
+{
+}
+
+string
+Attribute::OutputAttribute(string option)
+{
+	attr_generate = attribute + option;
+	return attr_generate;
+}
+
+BooleanAttribute::BooleanAttribute(string name, int prob)
+	: Attribute(name, prob)
+{
+}
+
+string
+BooleanAttribute::OutputAttributes()
+{
+	if(rnd_flipcoin(attribute_probability))
+		attr_generate = OutputAttribute("");
```

Comment:
This is weird, can you please remove one of `OutputAttribute`, `OutputAttributes` functions?

Reply:
Removed `OutputAttribute`.